src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTransformer.java

```diff
@@ -0,0 +1,235 @@
+    Set<RetryConfigDefinitionView.Name> retryNames = new HashSet<>();
+    for (Method method : context.getNonStreamingMethods()) {
+      MethodConfig conf = context.getMethodConfig(method);
+      retryNames.add(
+          RetryConfigDefinitionView.Name.create(
+              conf.getRetrySettingsConfigName(), conf.getRetryCodesConfigName()));
+    }
+
+    TreeMap<RetryConfigDefinitionView.Name, RetryConfigDefinitionView> retryDef = new TreeMap<>();
+    Map<String, ImmutableSet<Code>> retryCodesDef =
+        context.getInterfaceConfig().getRetryCodesDefinition();
+    Map<String, RetrySettings> retryParamsDef =
+        context.getInterfaceConfig().getRetrySettingsDefinition();
+    for (RetryConfigDefinitionView.Name name : retryNames) {
+      ImmutableSet<Code> codes = retryCodesDef.get(name.codes());
+      if (codes.isEmpty()) {
+        continue;
+      }
+      List<String> retryCodeNames = new ArrayList<>();
+      for (Code code : codes) {
+        retryCodeNames.add(namer.getStatusCodeName(code));
+      }
+      retryDef.put(
+          name,
+          RetryConfigDefinitionView.newBuilder()
+              .name(name)
+              .retryCodes(retryCodeNames)
+              .params(retryParamsDef.get(name.params()))
+              .build());
+    }
+    return new ArrayList<>(retryDef.values());
+  }
+
+  private Set<RetryConfigDefinitionView.Name> getRetryNames(SurfaceTransformerContext context) {
+    Set<RetryConfigDefinitionView.Name> retryNames = new HashSet<>();
+    for (Method method : context.getNonStreamingMethods()) {
+      MethodConfig conf = context.getMethodConfig(method);
+      retryNames.add(
+          RetryConfigDefinitionView.Name.create(
+              conf.getRetrySettingsConfigName(), conf.getRetryCodesConfigName()));
+    }
+    return retryNames;
+  }
+
+  private static final String EMPTY_PROTO_PKG = "github.com/golang/protobuf/ptypes/empty";
+
+  private void addXApiImports(SurfaceTransformerContext context, boolean hasRetry) {
+    ModelTypeTable typeTable = context.getTypeTable();
+    typeTable.saveNicknameFor("fmt;;;");
+    typeTable.saveNicknameFor("runtime;;;");
+    typeTable.saveNicknameFor("golang.org/x/net/context;;;");
+    typeTable.saveNicknameFor("google.golang.org/grpc;;;");
+    typeTable.saveNicknameFor("google.golang.org/grpc/metadata;;;");
+    typeTable.saveNicknameFor("github.com/googleapis/gax-go;gax;;");
+    typeTable.saveNicknameFor("google.golang.org/api/option;;;");
+    typeTable.saveNicknameFor("google.golang.org/api/transport;;;");
+
+    if (hasRetry) {
+      typeTable.saveNicknameFor("time;;;");
+      typeTable.saveNicknameFor("google.golang.org/grpc/codes;;;");
+    }
+    for (Method method : context.getNonStreamingMethods()) {
+      if (context.getMethodConfig(method).isPageStreaming()) {
+        typeTable.saveNicknameFor("math;;;");
+        break;
+      }
+    }
+    typeTable.getImports().remove(EMPTY_PROTO_PKG);
```

Comment:
I don't think this is something for "all" Go files. This might not need to be done in Apiary. This is mostly specific to how protobuf and Go interacts, but ModelTypeNameConverter is supposed to have a larger scope than that right?